Error when running "build MochaTest" on Windows

I'm still getting one error when building Fable - I've seen this before, but I'm not sure what could be causing this. My build might not be entirely clean because `build Clean` fails on Windows (paths too long), so I deleted old build folders by hand - but I may have forgotten some...

The error I'm getting is:

    cmd /C npm run test --

    > fable-tests@0.1.0 test c:\tomas\public\fsprojects\Fable\build\tests
    > mocha .

    module.js:327
        throw err;
        ^

    Error: Cannot find module './DllRef/DllRef/Util/Util'
      at Function.Module._resolveFilename (module.js:325:15)
      at Function.Module._load (module.js:276:25)
      at Module.require (module.js:353:17)
      at require (internal/module.js:12:17)
      at Object.<anonymous> (c:\tomas\public\fsprojects\Fable\build\tests\MiscTests.js:15:14)

--- a/src/fable/Fable.Compiler/Fable2Babel.fs
+++ b/src/fable/Fable.Compiler/Fable2Babel.fs
@@ -145,7 +145,7 @@ module Util =
                 match proj.ImportPath with
                 | Some importPath ->
                     let ext = Path.getExternalImportPath com ctx.fixedFileName importPath
-                    let rel = Path.getRelativePath proj.BaseDir file
+                    let rel = Path.getRelativeFileOrDirPath true proj.BaseDir false file
                     System.IO.Path.Combine(ext, rel)
                     |> Path.normalizePath
                     |> fun x -> System.IO.Path.ChangeExtension(x, null)

--- a/src/fable/Fable.Core/Util.fs
+++ b/src/fable/Fable.Core/Util.fs
@@ -149,14 +149,14 @@ module Path =
 
     /// Creates a relative path from one file or folder to another.
     /// from http://stackoverflow.com/a/340454/3922220
-    let getRelativePath fromPath toPath =
+    let getRelativeFileOrDirPath fromIsDir fromPath toIsDir toPath =
         // Add a dummy file to make it work correctly with dirs
-        let addDummyFile path =
-            if IO.Directory.Exists(path)
+        let addDummyFile isDir path =
+            if isDir
             then IO.Path.Combine(path, "dummy.txt")
             else path
-        let fromPath = addDummyFile fromPath
-        let toPath = addDummyFile toPath
+        let fromPath = addDummyFile fromIsDir fromPath
+        let toPath = addDummyFile toIsDir toPath
         let fromUri = Uri(fromPath)
         let toUri = Uri(toPath)
         if fromUri.Scheme <> toUri.Scheme then
@@ -169,6 +169,11 @@ module Path =
                             IO.Path.AltDirectorySeparatorChar,
                             IO.Path.DirectorySeparatorChar)  |> normalizePath
             | _ -> relativePath |> normalizePath
+
+    let getRelativePath fromPath toPath =
+        getRelativeFileOrDirPath 
+          (IO.Directory.Exists fromPath) fromPath 
+          (IO.Directory.Exists toPath) toPath
 
     let getExternalImportPath (com: ICompiler) (filePath: string) (importPath: string) =
         if not(importPath.StartsWith ".")

